### PR TITLE
CI/CD scripts

### PR DESCRIPTION
**Pull request by Vadim-Vorotilin**

### CODE REVIEW

**Review by mavimaec — APPROVED** [review]
Thanks @Vadim-Vorotilin , approving to unblock, just a little question below!

File: .github/workflows/publish.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        uses: actions/setup-node@v2
+        with:
+          node-version: '14.x'
+          registry-url: 'https://registry.npmjs.org'
+      - name: Install dependencies
+        run: npm install
+      - name: Build
+        run: npm run build
+      - name: Publish package on NPM dev 📦
+        run: npm run publish-dev
```
[the remark this answers is not in the document]
Comment (Vadim-Vorotilin) [reply .github/workflows/publish.yaml:23]: We'll need to have two separate workflows which would have different criteria of start. Now, this one will run if a release created. We can also unify these workflows using actions, like we did with the submodule libraries (g3d, bfast, etc.).
We could have (1) the dev publish workflow which would be triggered manually, (2) the official release workflow that would be triggered automatically when a release is created. But that would work if we create releases only for the official versions. If we create releases for the dev versions, both workflows would have to be triggered manually.

Alternatively, we could create a separate publish script that would analyze whether a dev or an official release should be published based on the name of the tag or something like that. What would be the best approach? Manual or automatic?

[the remark this answers is not in the document]
Comment (mavimaec) [reply .github/workflows/publish.yaml:23]: Here's my intuition:
- "dev" releases could occur automatically when a branch is merged into `develop`.
- official releases can be created when a release is created manually via GitHub Releases.

@vim-sroberge , what are your thoughts on this approach?

[the remark this answers is not in the document]
Comment (Vadim-Vorotilin) [reply .github/workflows/publish.yaml]: @vim-sroberge 
In the latest commit [90977bf](https://github.com/vimaec/vim-webgl-viewer/pull/202/commits/90977bfc7d27877f9db848b825f28a64b87fe37d) I've added another workflow .github/workflows/publish-dev.yaml
It automatically bumps the dev version and publishes the dev package. The workflow .github/workflows/publish.yaml now creates an official release once a repository release is created.

@mavimaec is that what you were talking about?

[the remark this answers is not in the document]
Comment (mavimaec) [reply .github/workflows/publish.yaml]: Yes @Vadim-Vorotilin that's what I was talking about, nice work!

**Pull request merged**